core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -850,6 +851,60 @@ public ExecResult execInContainer(String... command)
         return execInContainer(UTF8, command);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void copyFileToContainer(MountableFile mountableLocalFile, String containerPath) throws IOException, InterruptedException {
+
+        this.dockerClient
+                .copyArchiveToContainerCmd(this.containerId)
+                .withHostResource(mountableLocalFile.getResolvedPath())
+                .withRemotePath(containerPath)
+                .exec();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void copyFileFromContainer(String containerPath, String destinationPath) throws IOException, InterruptedException {
+        try (final InputStream response = this.dockerClient
+                .copyArchiveFromContainerCmd(this.containerId, containerPath)
+                .exec()) {
+            try (final TarArchiveInputStream tarInputStream = new TarArchiveInputStream(response)) {
```

Comment:
no need for the nested `try`s, just wrap `.exec()` result into `TarArchiveInputStream`